### PR TITLE
Feature/mip install log

### PR DESCRIPTION
**Pull request by jemten**

### CODE REVIEW

File: lib/MIP/Language/Shell.pm [hunk 1]
```diff
@@ -158,7 +155,7 @@ sub create_bash_file {
 
     if ( ( defined $log ) && ($log) ) {
 
-        say $log->info( q{Created bash file: '} . catfile($file_name),
+        $log->info( q{Created bash file: '} . catfile($file_name),
             $SINGLE_QUOTE );
```
Comment (henrikstranneheim) [inline lib/MIP/Language/Shell.pm]: Don't you want a $NEWLINE here?

File: lib/MIP/Log/MIP_log4perl.pm [hunk 2]
```diff
@@ -141,4 +141,66 @@ EOF
     return $config;
 }
 
+sub retrieve_log {
+
+## retrieve_log
```
Comment (henrikstranneheim) [inline lib/MIP/Log/MIP_log4perl.pm]: Remove

File: mip_install.pl [hunk 3]
```diff
@@ -2,35 +2,36 @@
...
-use warnings qw( FATAL utf8 );
+use warnings qw{ FATAL utf8 };
 use utf8;
-use open qw( :encoding(UTF-8) :std );
-use charnames qw( :full :short );
+use open qw{ :encoding(UTF-8) :std };
+use charnames qw{ :full :short };
 use Carp;
-use English qw(-no_match_vars);
+use English qw{-no_match_vars};
```
Comment (henrikstranneheim) [inline mip_install.pl]: qw{-no_match_vars} -> qw{ -no_match_vars }

File: mip_install.pl [hunk 4]
```diff
@@ -2,35 +2,36 @@
...
 use Params::Check qw[check allow last_error];
```
Comment (henrikstranneheim) [inline mip_install.pl]: qw{}

File: mip_install.pl [hunk 5]
```diff
@@ -2,35 +2,36 @@
...
 $Params::Check::PRESERVE_CASE = 1;    #Do not convert to lower case
```
Comment (henrikstranneheim) [inline mip_install.pl]: Remove if you dare, :)!

File: mip_install.pl (hunk 3 above)
Comment (henrikstranneheim) [reply mip_install.pl]: Add space in the qw{}. Throughout

File: mip_install.pl [hunk 6]
```diff
@@ -248,8 +266,8 @@
 my $file_name_path = catfile( cwd(), q{mip.sh} );
 
 open $FILEHANDLE, q{>}, $file_name_path
-  or
-  croak( q{Cannot write to '} . $file_name_path . q{' :} . $OS_ERROR . "\n" );
+  or $log->logcroak(
+    q{Cannot write to '} . $file_name_path . q{' :} . $OS_ERROR . "\n" );
```
Comment (henrikstranneheim) [inline mip_install.pl]: $SPACE and $NEWLINE

File: mip_install.pl [hunk 7]
```diff
@@ -259,10 +277,11 @@
         remove_dir  => catfile( cwd(), q{.MIP} ),
```
Comment (henrikstranneheim) [inline mip_install.pl]: $DOT . q{MIP}

File: mip_install.pl [hunk 8]
```diff
@@ -2578,44 +2601,56 @@ sub check_conda_bin_file_exists {
...
+                      . $SPACE
+                      . $program_name
+                      . $SPACE
+                      . $program_version );
                 return 1;
             }
-            print STDERR 'Writting install instructions for ' . $program_name,
-              "\n";
+            $log->info(
+                q{Writting install instructions for} . $SPACE . $program_name );
```
Comment (henrikstranneheim) [inline mip_install.pl]: $NEWLINE?

File: mip_install.pl [hunk 9]
```diff
@@ -2578,44 +2601,56 @@ sub check_conda_bin_file_exists {
...
+                  . q{in miniconda directory:}
+                  . $SPACE
+                  . catdir( $parameter_href->{conda_prefix_path}, q{bin} ) );
 
             if ( $parameter_href->{noupdate} ) {
 
-                print STDERR 'Skipping writting installation process for '
-                  . $program_name . q{ }
-                  . $program_version, "\n";
+                $log->info( q{Skipping writting installation process for}
```
Comment (henrikstranneheim) [inline mip_install.pl]: $NEWLINE?

File: mip_install.pl [hunk 10]
```diff
@@ -2578,44 +2601,56 @@ sub check_conda_bin_file_exists {
...
-              . $program_name
-              . ' in miniconda directory: '
-              . catdir( $parameter_href->{conda_prefix_path}, 'bin' ), "\n";
+            $log->info( q{Found}
+                  . $SPACE
+                  . $program_name
+                  . $SPACE
+                  . q{in miniconda directory:}
+                  . $SPACE
+                  . catdir( $parameter_href->{conda_prefix_path}, q{bin} ) );
```
Comment (henrikstranneheim) [inline mip_install.pl]: $NEWLINE?

File: mip_install.pl (hunk 5 above)
Comment (jemten) [reply mip_install.pl]: challenge accepted :)

File: mip_install.pl [hunk 11]
```diff
@@ -52,7 +52,7 @@
 my %parameter;
 
 ## Log
-$parameter{log_file} = q{mip_install.log};
+#$parameter{log_file} = q{mip_install.log};
```
Comment (henrikstranneheim) [inline mip_install.pl]: Remove

File: mip_install.pl [hunk 12]
```diff
@@ -206,7 +206,17 @@
...
 
-## Create log object for mip_install
+## Get local time
+my $date_time       = localtime;
+my $date_time_stamp = $date_time->datetime;
+
+## Create default log name
+if ( not $parameter{log_file} ) {
+    $parameter{log_file} =
+      catfile( q{mip_install_} . $date_time_stamp . $DOT . q{log} );
```
Comment (henrikstranneheim) [inline mip_install.pl]: $UNDERSCORE

**Comment by henrikstranneheim** [comment]
Fantastic!

**Pull request merged**